packages/lodestar/src/tasks/tasks/archiveStates.ts

```diff
@@ -2,53 +2,108 @@
  * @module tasks
  */
 
-import {ITask} from "../interface";
-import {IBeaconDb} from "../../db/api";
-import {IBeaconConfig} from "@chainsafe/lodestar-config";
 import {ILogger} from "@chainsafe/lodestar-utils";
 import {phase0} from "@chainsafe/lodestar-types";
 import {TreeBacked} from "@chainsafe/ssz";
+import {IBeaconConfig} from "@chainsafe/lodestar-config";
+import {computeEpochAtSlot} from "@chainsafe/lodestar-beacon-state-transition";
+import {IBeaconDb} from "../../db/api";
 import {IBeaconChain} from "../../chain";
 
-export interface IArchiveStatesModules {
+/**
+ * Minimum number of epochs between archived states
+ */
+const PERSIST_STATE_EVERY_EPOCHS = 1024;
+/**
+ * Minimum number of epochs between single temp archived states
+ * These states will be pruned once a new state is persisted
+ */
+const PERSIST_TEMP_STATE_EVERY_EPOCHS = 32;
+
+export type StatesArchiverModules = {
   chain: IBeaconChain;
   db: IBeaconDb;
   logger: ILogger;
-}
+};
 
 /**
  * Archives finalized states from active bucket to archive bucket.
  *
  * Only the new finalized state is stored to disk
  */
-export class ArchiveStatesTask implements ITask {
+export class StatesArchiver {
+  private readonly config: IBeaconConfig;
   private readonly chain: IBeaconChain;
   private readonly db: IBeaconDb;
   private readonly logger: ILogger;
-  private readonly config: IBeaconConfig;
-
-  private finalized: phase0.Checkpoint;
 
-  public constructor(config: IBeaconConfig, modules: IArchiveStatesModules, finalized: phase0.Checkpoint) {
+  public constructor(config: IBeaconConfig, modules: StatesArchiverModules) {
+    this.config = config;
     this.chain = modules.chain;
     this.db = modules.db;
     this.logger = modules.logger;
-    this.config = config;
-    this.finalized = finalized;
   }
 
-  public async run(): Promise<void> {
-    // store the state of finalized checkpoint
-    const stateCache = this.chain.checkpointStateCache.get(this.finalized);
+  /**
+   * Persist states every some epochs to
+   * - Minimize disk space, storing the least states possible
+   * - Minimize the sync progress lost on unexpected crash, storing temp state every few epochs
+   *
+   * At epoch `x` there will be states peristed at intervals of `MIN_EPOCHS_PER_DB_STATE`
+   * and one at `MIN_EPOCHS_PER_DB_STATE / LATEST_STATE_FACTOR`
+   * ```
+   *    |         |       |    |
+   * x-1024*2  x-1024   x-32   x
```

Comment:
can you add a clarifying comment to say that what 1024 and 32 represent here (e.g. if they are MIN_EPOCHS_PER_DB_STATE=1024 and LATEST_STATE_FACTOR=32) since it may not be that value in the future (or maybe it'll even be cli configurable)?